crates/circuit/src/dag_circuit.rs

```diff
@@ -4531,6 +5038,162 @@ impl Default for DAGCircuit {
+        instr: &'a PackedInstruction,
+    ) -> Option<ControlFlowView<'a, DAGCircuit>> {
+        let OperationRef::ControlFlow(control) = instr.op.view() else {
+            return None;
+        };
+        Some(match control {
+            ControlFlow::Box { duration, .. } => ControlFlowView::Box(
+                duration.as_ref(),
+                self.blocks.get(instr.blocks_view()[0].index()).unwrap(),
+            ),
+            ControlFlow::BreakLoop { .. } => ControlFlowView::BreakLoop,
+            ControlFlow::ContinueLoop { .. } => ControlFlowView::ContinueLoop,
+            ControlFlow::ForLoop {
+                indexset,
+                loop_param,
+                ..
+            } => ControlFlowView::ForLoop {
+                indexset: indexset.as_slice(),
+                loop_param: loop_param.as_ref(),
+                body: self.blocks.get(instr.blocks_view()[0].index()).unwrap(),
+            },
+            ControlFlow::IfElse { condition, .. } => ControlFlowView::IfElse {
+                condition,
+                true_body: self.blocks.get(instr.blocks_view()[0].index()).unwrap(),
+                false_body: instr
+                    .blocks_view()
+                    .get(1)
+                    .map(|b| self.blocks.get(b.index()).unwrap()),
+            },
+            ControlFlow::Switch {
+                target, label_spec, ..
+            } => {
+                let cases_specifier = label_spec
+                    .iter()
+                    .zip(
+                        instr
+                            .blocks_view()
+                            .iter()
+                            .map(|case| self.blocks.get(case.index()).unwrap()),
+                    )
+                    .collect();
+                ControlFlowView::Switch {
+                    target,
+                    cases_specifier,
+                }
+            }
+            ControlFlow::While { condition, .. } => ControlFlowView::While {
+                condition,
+                body: self.blocks.get(instr.blocks_view()[0].index()).unwrap(),
+            },
+        })
+    }
+
+    /// Build a reference to the Python-space operation object (the `Gate`, etc) packed into an
+    /// instruction.  This may construct the reference if the `Instruction` is a standard
+    /// gate or instruction with no already stored operation.
+    ///
+    /// A standard-gate or standard-instruction operation object returned by this function is
+    /// disconnected from the dag; updates to its parameters, label, duration, unit
+    /// and condition will not be propagated back.
+    ///
+    /// Panics if `node` does not refer to an operation.
+    #[inline]
+    pub fn unpack_py_op(&self, py: Python, node: NodeIndex) -> PyResult<Py<PyAny>> {
+        let instr = self.dag[node].unwrap_operation();
+        self.unpack_py_op_inner(py, instr)
+    }
+
+    fn unpack_py_op_inner(&self, py: Python, instr: &PackedInstruction) -> PyResult<Py<PyAny>> {
```

Comment:
Same comment as above and on `CircuitData`: can we make `unpack_py_op` the native format?

Reply:
Done in 761bdcd836.